app/src/selfRelease/java/org/radarcns/detail/OneTimeScheduleWorker.kt

```diff
@@ -0,0 +1,84 @@
+package org.radarcns.detail
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import org.radarcns.detail.UpdateScheduledService.Companion.UPDATE_VERSION_NAME_KEY
+import org.radarcns.detail.UpdateScheduledService.Companion.UPDATE_VERSION_URL_KEY
+import kotlin.random.Random
+
+class OneTimeScheduleWorker(
+    private val context: Context,
+    workerParams: WorkerParameters
+) : Worker(context, workerParams) {
+
+    override fun doWork(): Result {
+        val url =  inputData.getString(UPDATE_VERSION_URL_KEY)
+        val versionName =  inputData.getString(UPDATE_VERSION_NAME_KEY)
+
+        val intent = Intent(context, UpdatesActivity::class.java).apply {
+            flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
+        }
+        intent.putExtra(FROM_NOTIFICATION_KEY, true)
+        intent.putExtra(UPDATE_VERSION_NAME_KEY, versionName)
+        intent.putExtra(UPDATE_VERSION_URL_KEY, url)
```

Comment:
You can move the intent sets inside the apply
```suggestion
            flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            putExtra(FROM_NOTIFICATION_KEY, true)
            putExtra(UPDATE_VERSION_NAME_KEY, versionName)
            putExtra(UPDATE_VERSION_URL_KEY, url)
        }
```